### ISSUE TITLE
FAILED HttpServerBug21008Spec

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#493]
Probably related to #396 or #448. Probably doesn't make sense to work on it as long as those have not been fixed.

https://jenkins.akka.io:8498/job/pr-validator-akka-http/238/testReport/junit/akka.http.impl.engine.server/HttpServerBug21008Spec/The_HttpServer_should_not_cause_internal_graph_failures_when_consuming_a__100_Continue__entity_triggers_a_failure/

```
akka.http.impl.engine.server.HttpServerBug21008Spec.The HttpServer should not cause internal graph failures when consuming a `100 Continue` entity triggers a failure

org.scalatest.exceptions.TestFailedException: HttpServerBluePrint.ControllerStage: requirement failed: Cannot pull closed port (requestParsingIn),
inside HttpRequest(HttpMethod(POST),http://example.com/,List(Host: example.com, Expect: 100-continue, Timeout-Access: <function1>),HttpEntity.Chunked(application/octet-stream),HttpProtocol(HTTP/1.1))
	at org.scalatest.Assertions$class.newAssertionFailedException(Assertions.scala:528)
	at akka.testkit.AkkaSpec.newAssertionFailedException(AkkaSpec.scala:57)
	at org.scalatest.Assertions$class.fail(Assertions.scala:1089)
	at akka.testkit.AkkaSpec.fail(AkkaSpec.scala:57)
	at akka.http.impl.engine.server.HttpServerBug21008Spec$$anonfun$1$$anonfun$apply$mcV$sp$1$$anonfun$apply$1$$anon$1$$anonfun$2.applyOrElse(HttpServerBug21008Spec.scala:70)
	at akka.http.impl.engine.server.HttpServerBug21008Spec$$anonfun$1$$anonfun$apply$mcV$sp$1$$anonfun$apply$1$$anon$1$$anonfun$2.applyOrElse(HttpServerBug21008Spec.scala:38)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at org.scalatest.Inside$class.inside(Inside.scala:128)
	at akka.http.impl.engine.server.HttpServerBug21008Spec.inside(HttpServerBug21008Spec.scala:18)
	at akka.http.impl.engine.server.HttpServerBug21008Spec$$anonfun$1$$anonfun$apply$mcV$sp$1$$anonfun$apply$1$$anon$1.<init>(HttpServerBug21008Spec.scala:38)
	at akka.http.impl.engine.server.HttpServerBug21008Spec$$anonfun$1$$anonfun$apply$mcV$sp$1$$anonfun$apply$1.apply(HttpServerBug21008Spec.scala:28)
	at akka.http.impl.engine.server.HttpServerBug21008Spec$$anonfun$1$$anonfun$apply$mcV$sp$1$$anonfun$apply$1.apply(HttpServerBug21008Spec.scala:28)
	at akka.stream.testkit.Utils$.assertAllStagesStopped(Utils.scala:25)
	at akka.http.impl.engine.server.HttpServerBug21008Spec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(HttpServerBug21008Spec.scala:28)
	at akka.http.impl.engine.server.HttpServerBug21008Spec$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(HttpServerBug21008Spec.scala:28)
	at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:1078)
	at org.scalatest.TestSuite$class.withFixture(TestSuite.scala:196)
	at akka.testkit.AkkaSpec.withFixture(AkkaSpec.scala:57)
	at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:1075)
	at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
	at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:1088)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
	at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:1088)
	at akka.testkit.AkkaSpec.runTest(AkkaSpec.scala:57)
	at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
	at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1147)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:396)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:373)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:410)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:384)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:379)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
	at org.scalatest.WordSpecLike$class.runTests(WordSpecLike.scala:1147)
	at akka.testkit.AkkaSpec.runTests(AkkaSpec.scala:57)
	at org.scalatest.Suite$class.run(Suite.scala:1147)
	at akka.testkit.AkkaSpec.org$scalatest$WordSpecLike$$super$run(AkkaSpec.scala:57)
	at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
	at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1192)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
	at org.scalatest.WordSpecLike$class.run(WordSpecLike.scala:1192)
	at akka.testkit.AkkaSpec.org$scalatest$BeforeAndAfterAll$$super$run(AkkaSpec.scala:57)
	at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:213)
	at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:210)
	at akka.testkit.AkkaSpec.run(AkkaSpec.scala:57)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:314)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:472)
	at sbt.TestRunner.runTest$1(TestFramework.scala:76)
	at sbt.TestRunner.run(TestFramework.scala:85)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:202)
	at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:185)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:202)
	at sbt.TestFunction.apply(TestFramework.scala:207)
	at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
	at sbt.Tests$$anonfun$9.apply(Tests.scala:216)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by jonas** [comment]
Else we could reopen (and retitle) #19 since it looks like bumping the leeway timeout in #479 didn't fix the failure.

**Comment by jrudolph** [comment]
If I'm not mistaken this is not a simple timing issue but a real race condition.

**Comment by jrudolph** [comment]
You are right. I see what you mean. Thanks for the pointer to the old ones.

**Comment by ktoso** [comment]
For reference, another one here: https://jenkins.akka.io:8498/job/pr-validator-akka-http/219/consoleFull

Let's not re-open tickets but have this one and point to other instances of the failure. Makes tracking what happened when easier (also, if we thought we fixed it in a release etc)

**Comment by jrudolph** [comment]
Hopefully fixed by #869. Marking this as duplicate of #516.
